File: app/src/main/java/com/nordokod/scio/old/entity/Answer.java
package com.nordokod.scio.old.entity;

import java.io.Serializable;
public class Answer implements Serializable {
    private String answer;
    private boolean correct;
    public Answer(String answer, boolean correct) {
        this.answer = answer;
        this.correct = correct;
    }

    public String getAnswer() {
        return answer;
    }
    public boolean isCorrect() {
        return correct;
    }
}

File: app/src/main/java/com/nordokod/scio/kt/model/entity/Answer.kt
package com.nordokod.scio.kt.model.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

sealed class Answer {
    @Entity
    data class OpenAnswer(
            @PrimaryKey(autoGenerate = true) var id: Long = 0,
            var idQuestion: Long = 0,
            var answer: String = ""
    ) : Answer()

    @Entity
    data class TrueFalseAnswer(
            @PrimaryKey(autoGenerate = true) var id: Long = 0,
            var idQuestion: Long = 0,
            var answer: Boolean = true
    ) : Answer()

    @Entity
    data class MultipleChoiceAnswer(
            @PrimaryKey(autoGenerate = true) var id: Long = 0,
            var idQuestion: Long = 0,
            var answer: String = "",
            var isCorrect: Boolean = true
    ) : Answer()
}